src/Viewer/services/ActionHandler.js

```diff
@@ -0,0 +1,222 @@
+import CLP_WORKER_PROTOCOL from "./CLP_WORKER_PROTOCOL";
+import FileManager from "./decoder/FileManager";
+import {isBoolean, isNumeric} from "./decoder/utils";
+
+// TODO: Move decompressing of file from FileManager to ActionHandler.
+//  When there are multiple IRStreams in a single file, this will allow
+//  the creation of multiple FileManagers. When this feature is implemented,
+//  the action handler will contain multiple IRStreams and user can be prompted
+//  for which IRStream they want to load in the current viewer.
+
+/**
+ * Manages all the actions that can be executed on provided file. In the future,
+ * this action handler will also be used to execute jobs on file data.
+ */
+class ActionHandler {
+    /**
+     * Initialize the action handler.
+     */
+    constructor () {
+        this._logFile = null;
+    }
+
+    /**
+     * Creates a new FileManager object and initiates the download.
+     *
+     * @param {String|File} fileInfo
+     * @param {boolean} prettify
+     * @param {Number} logEventIdx
+     * @param {Number} pageSize
+     */
+    loadFile (fileInfo, prettify, logEventIdx, pageSize) {
+        this._logFile = new FileManager(fileInfo, prettify,
+            logEventIdx, pageSize, this._loadingMessageCallback, this._updateStateCallback,
+            this._updateLogsCallback, this._updateFileInfoCallback);
+        this._logFile.decompressAndLoadFile();
+    }
+
+    /**
+     * Filters the events at the given verbosity and rebuilds the pages.
+     *
+     * @param {number} desiredVerbosity
+     */
+    changeVerbosity (desiredVerbosity) {
+        if (!isNumeric(desiredVerbosity)) {
+            throw (new Error("Invalid verbosity provided."));
+        }
+        this._logFile._state.verbosity = desiredVerbosity;
```

Comment:
Shouldn't access FileManager's private state from outside the class. A lot of that in this class.

Reply:
Generally you don't want to make your class variables public because that makes it a lot harder to track modifications to those variables compared to using getters and setters. Anyway, since you've made the change already, I guess let's wait for all the refactoring changes and then we can revisit this.